resources/lib/player/players.py

```diff
@@ -144,18 +144,28 @@ def _set_external_ids(self, tmdb_type, tmdb_id, season, episode, details, requir
             details.set_details(details=self.details_ext_ids, reverse=True)
         return set_detailed_item(tmdb_type, tmdb_id, season, episode, details=details) or {}
 
-    def _set_provider_priority(self):
-        try:  # Check if players are listed in providers and bump priority if found
-            providers = self.details.infoproperties['providers']
-            providers = providers.split(' / ')
-            for k, v in self.players.items():
-                if 'provider' not in v or v['provider'] not in providers:
-                    v['priority'] = v.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
-                    continue
-                v['priority'] = providers.index(v['provider']) + 1  # Add 1 because sorted() puts 0 index last
-                v['is_provider'] = True
-        except (KeyError, AttributeError):
-            pass
+    def _get_prioritised_players(self):
+        try:
+            providers = self.details.infoproperties.get('providers', '')
+        except AttributeError:
+            providers = ''
+        providers = tuple(providers.split(' / '))
+
+        def _set_priority(item, _providers=providers):
+            file, player = item
+            player_provider = player.get('provider')
```

Comment:
I think might be a good idea here to make sure if a blank string is returned it gets converted to None type:
`player_provider = player.get('provider') or None`

That way we avoid `player_provider = ''` matching `_providers = ''.split(' / ')`

Reply:
Yep, makes sense. Updated in [386bb11](https://github.com/jurialmunkey/plugin.video.themoviedb.helper/pull/965/commits/386bb11f926b2ccb24b5d4b11a8d8bdcbc270eaf)